src/background/services/history/index.ts

```diff
@@ -0,0 +1,66 @@
+import { getEnabledFeatures } from "@src/config/features";
+import { Operation, OperationType } from "@src/types";
+
+import type { OperationOptions, OperationFilter } from "./types";
+
+import LockService from "../lock";
+import SimpleStorage from "../simpleStorage";
+
+const HISTORY_KEY = "@@HISTORY@@";
+
+export * from "./types";
+
+export default class HistoryService {
+  private static INSTANCE: HistoryService;
+
+  private historyStore: SimpleStorage;
+
+  private lockService: LockService;
+
+  private operations: Operation[];
+
+  private constructor() {
+    this.historyStore = new SimpleStorage(HISTORY_KEY);
+    this.lockService = LockService.getInstance();
+    this.operations = [];
+  }
+
+  public static getInstance(): HistoryService {
+    if (!HistoryService.INSTANCE) {
+      HistoryService.INSTANCE = new HistoryService();
+    }
+
+    return HistoryService.INSTANCE;
+  }
+
+  public loadOperations = async (): Promise<Operation[]> => {
+    const features = getEnabledFeatures();
+    const serializedOperations = await this.historyStore
+      .get<string>()
+      .then((serialized) => (serialized ? (JSON.parse(serialized) as Operation[]) : []));
+
+    this.operations = serializedOperations.filter(({ identity }) =>
+      !features.RANDOM_IDENTITY ? identity.metadata.identityStrategy !== "random" : true,
+    );
+
+    return this.operations;
+  };
+
+  public getOperations = (filter?: Partial<OperationFilter>): Operation[] =>
+    this.operations.filter((operation) => (filter?.type ? operation.type === filter.type : true));
+
+  public trackOperation = async (type: OperationType, { identity }: OperationOptions): Promise<void> => {
```

Comment:
I think it's not an optimized way to deal with the storage, we already save the `Idenitty Metadata` in the identity commitment, we dont have to re-store it here again. We instead can only use a `commitment` easier to be a pointer to the Identity storage.

Reply:
I don't think it will work with all the operation. For instance, we can't get deleted identity from store because there is no such identity. In addition, it will help to track identity changes (like name). That's the reason I duplicate this data for history.